robottelo/ui/base.py

```diff
@@ -115,21 +115,22 @@ def search(self, element_name):
         searchbox.send_keys(u'{0} = {1}'.format(
             search_key, escape_search(element_name)))
         self.click(search_button_locator)
-
+        # Make sure that found element is returned no matter it described by
+        # its own locator or common one (locator can transform depending on
+        # element name length)
+        delay = 0
         strategy, value = element_locator
-        # If foreman entity and it's length more than 32 chars, and not in
-        # exceptions list use the common locator.
-        if (
-                not self.is_katello and
-                len(element_name) > 32 and
-                type(self).__name__ not in SEARCH_EXCEPTIONS_LIST):
-            strategy, value = common_locators['select_filtered_entity']
-        # Return found element
-        element = self.wait_until_element(
-            (strategy, value % element_name),
-            timeout=self.result_timeout,
-        )
-        return element
+        strategy2, value2 = common_locators['select_filtered_entity']
+        while delay < self.result_timeout:
+            element = self.find_element((strategy, value % element_name))
+            element2 = self.find_element((strategy2, value2 % element_name))
```

Comment:
What about moving this after the if block below? This way if the element is not None it will not perform this action.